src/search_renderer.cpp

```diff
@@ -93,37 +93,42 @@ std::string SearchRenderer::getHtml()
   kainjow::mustache::data pages{kainjow::mustache::data::type::list};
 
   auto resultStart = mp_searcher->getResultStart();
-  auto resultEnd = mp_searcher->getResultEnd();
-  auto resultCountPerPage = resultEnd - resultStart;
+  auto pageLength = this->pageLength;
   auto estimatedResultCount = mp_searcher->getEstimatedResultCount();
   auto currentPage = 0U;
   auto pageStart = 0U;
   auto pageEnd = 0U;
   auto lastPageStart = 0U;
-  if (resultCountPerPage) {
-    currentPage = resultStart/resultCountPerPage;
+  if (pageLength) {
+    currentPage = resultStart/pageLength;
     pageStart = currentPage > 4 ? currentPage-4 : 0;
     pageEnd = currentPage + 5;
-    if (pageEnd > estimatedResultCount / resultCountPerPage) {
-      pageEnd = estimatedResultCount / resultCountPerPage;
+    if (pageEnd > estimatedResultCount / pageLength) {
+      pageEnd = estimatedResultCount / pageLength;
     }
-    if (estimatedResultCount > resultCountPerPage) {
-      lastPageStart = static_cast<int>(round(estimatedResultCount/resultCountPerPage)) * resultCountPerPage;
+    if (estimatedResultCount > pageLength) {
+      lastPageStart = static_cast<int>(round(estimatedResultCount/pageLength)) * pageLength;
     }
+    if(pageEnd*pageLength==estimatedResultCount && pageEnd>0)pageEnd--;
```

Comment:
I feel that the `pageEnd` management logic can be simplified even further by taking advantage of the implicit relation between `pageEnd` and `lastPageStart`

Reply:
@veloman-yunkan  ok let me take a look I have some more ideas to apply to make code shorter and more efficient, I will make a final review attempt tomorrow after applying all the changes by then if you have any more suggestions please let me now :)

Reply:
@veloman-yunkan I guess pageEnd and lastPageStart are not always related because pageEnd is the page which is at max  +5 of current page but lastPageStart is start of the last page of whole result.